modules/caddyhttp/app.go

```diff
@@ -226,6 +226,8 @@ func (app *App) Provision(ctx caddy.Context) error {
 		// route handler so that important security checks are done, etc.
 		primaryRoute := emptyHandler
 		if srv.Routes != nil {
+			// inject the server name for observability purposes
+			ctx.Context = contextWithServerName(ctx.Context, srvName)
```

Comment:
I'll need to look at this on a second pass and think if this is really the best way to do this.

Reply:
I tried a few different ways, and injecting the server name into the context seems to be really the only option without breaking API. But open to suggestions 😉

Reply:
I guess this is fine. It's very internal anyway so we can change it later.